### ISSUE TITLE
AWI-CM 2: Monthly restarts: Last month/restart will not be executed.

### ISSUE THREAD

**Issue by nwieters** [esm-tools/esm_tools#20]
The following yaml runscript for AWI-CM 2 will stop before last month. It will only run 11 months allthough final date is 2002-01-01.

  1 general:
  2     setup_name: "awicm"
  3     compute_time: "00:15:00"
  4     initial_date: "2000-01-01"
  5     final_date: "2002-01-01"
  6     base_dir: "/work/ollie/nwieters/ESM/Test_esm_tools/esm_experiments/"
  7     nmonth: 1
  8     nyear: 0
  9 
 10 awicm:
 11     version: "2.0"
 12     postprocessing: false
 13     scenario: "PI-CTRL"
 14     model_dir: "/work/ollie/nwieters/ESM/Test_esm_tools/awicm-2.0/"
 15 
 16 fesom:
 17     version: "2.0"
 18     res: "CORE2"
 19     pool_dir: "/work/ollie/pool/FESOM/"
 20     mesh_dir: "/work/ollie/dsidoren/input/fesom2.0/meshes/mesh_CORE2_finaltopo_mean/"
 21     restart_rate: 1
 22     restart_unit: "m"
 23     restart_first: 1
 24     lresume: 0

**Comment by nwieters** [comment]
```
  4     initial_date: "2000-01-01"
  5     final_date: "2000-02-01"
```

will result in two runs:
run_20000101-20000131  
run_20000201-20000229

**Comment by nwieters** [comment]
if 
```
  4     initial_date: "2000-01-01"
  5     final_date: "2000-12-31"
```
I get only
run_20000101-20000131
run_20000201-20000229
run_20000301-20000331
run_20000401-20000430
run_20000501-20000531
run_20000601-20000630
run_20000701-20000731
run_20000801-20000831
run_20000901-20000930
run_20001001-20001031
run_20001101-20001130

**Comment by nwieters** [comment]
This seems to be fixed with release 4.2.1 (or previouse ones).
